Makefile.am

```diff
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 AM_CPPFLAGS = -I$(top_srcdir)/libass
-AM_CFLAGS = -std=gnu99 -Wall -Wextra -Wno-sign-compare -Wno-unused-parameter \
+AM_CFLAGS = -Wall -Wextra -Wno-sign-compare -Wno-unused-parameter \
```

Comment:
No longer requesting an explicit standard version has undesireable effects:
 - our expected baseline is no longer documented at all
 - we now depend on compiler and/or autotools defaults which might change at any time and differ across compilers
 - with the current C11 default we’ll no longer test our actual default in CI, nor on developer machines
 
 This makes it likely we’ll accidentally break our supposed baseline at some point by inadvertenly using a newer feature *(see e.g. what happened with an `{}` between 0.17.1 and 0.17.2 because compilers didn’t warn about it)*.
 
 On the other hand, the reason for dropping this is some libcs otherwise hiding atomics; which libcs do that?
 
I think instead of just dropping the standard version we should do one of the following *(ordered by my personal preference)*:
- continue to explicitly request GNU99 **if** no libcs actually hide atomics
- raise our baseline to C11 and explicitly request GNU/ISO C11
- explicitly request _either_ GNU/ISO C11 _or_ GNU/ISO C99 and make sure at least one CI job always uses C99
- drop the explicit version like now, but ensure one CI job explicitly runs with our baseline version
 
*(Having a meson buildsystem and meson not _yet_ being changed actually helps out here, since it still explicitly requests ISO C99, but once meson will change too so this is only temporary)*

Reply:
`AC_PROG_CC` automatically selects the latest known non-strict-conformance (i.e. `gnu`) standard version the compiler supports. If we want, we can check `ac_prog_cc_stdc` and warn if it's `c89`, but I don't really see the need; if an older compiler fails on our code due to not supporting the minimum required standard, the user will be notified by, uh… compilation failing (which is already what happens now, just slightly faster via `-std=gnu99` being unrecognized). I think the only actual potential issue here is that, say, a future C version could come along and remove something we use? But generally the C standard provides years of deprecation notice, so I'm not really worried about that either.

Regardless of whether atomics are hidden in C99 mode, I see no benefit in unconditionally explicitly requesting an older version than what we're using features from.

Running a build with an explicit `-std=gnu[earliest supported version]` seems reasonable, though I think it'd be more useful to do one with the earliest *gcc version* we're targeting (which is a much more meaningful test than using a newer compiler in an older mode).

Documenting the minimum requirements somewhere seems reasonable (README?); as of this PR it's "99, or 11 for threading", and I expect there's a good chance it'll move to "11, unless we have an explicit compat shim for your system" in the near future (assuming we go forwards with requiring atomics for publicly-refcounted frames).